docs/Homebrew-homebrew-core-Merge-Checklist.md

```diff
@@ -1,22 +1,128 @@
-request changes from a contributor or help them out based on their comfort and
-previous contributions. Remember, as a team we
-[Prioritise Maintainers Over Users](Maintainers-Avoiding-Burnout.md) to avoid
-burnout.
+Don’t `merge` unclean branches. So if someone is still learning `git` and
+their branch is filled with nonsensical merges, then `rebase` and squash
+the commits. Our main branch history should be useful to other people,
+not confusing.
 
-This is a more practical checklist; it should be used after you get familiar with
-[Maintainer Guidelines](Maintainer-Guidelines.md).
+Here’s a flowchart for managing a PR which is ready to merge:
 
-## Checklist
+![Flowchart for managing pull requests](assets/img/docs/managing-pull-requests.drawio.svg)
 
-Check for:
+Only one maintainer is necessary to approve and merge the addition of a new or updated formula which passes CI. However, if the formula addition or update proves controversial the maintainer who adds it will be expected to answer requests and fix problems that arise with it in future.
+
+### How to merge without bottles
+
+Here are guidelines about when to use squash & merge versus rebase & merge. These options should only be used with PRs where bottles are not needed or affected.
+
+| | PR modified a single formula | PR modifies multiple formulae |
+|---|---|---|
+| **Commits look good** | rebase & merge _or_ squash & merge | rebase & merge |
+| **Commits need work** | squash & merge | manually merge using the command line |
+
+## Testing
+
+We need to at least check that it builds. Use the [Brew Test Bot](Brew-Test-Bot.md) for this.
+
+Verify the formula works if possible. If you can’t tell (e.g. if it’s a
+library) trust the original contributor, it worked for them, so chances are it
+is fine. If you aren’t an expert in the tool in question, you can’t really
+gauge if the formula installed the program correctly. At some point an expert
+will come along, cry blue murder that it doesn’t work, and fix it. This is how
+open source works. Ideally, request a `test do` block to test that
+functionality is consistently available.
+
+If the formula uses a repository, then the `url` parameter should have a
+tag or revision. `url`s have versions and are stable (not yet
+implemented!).
+
+Don't merge any formula updates with failing `brew test`s. If a `test do` block
+is failing it needs to be fixed. This may involve replacing more involved tests
+with those that are more reliable. This is fine: false positives are better than
+false negatives as we don't want to teach maintainers to merge red PRs. If the
+test failure is believed to be due to a bug in `Homebrew/brew` or the CI system,
+that bug must be fixed, or worked around in the formula to yield a passing test,
+before the PR can be merged.
+
+## Duplicates
+
+We now accept stuff that comes with macOS as long as it uses `keg_only :provided_by_macos` to be keg-only by default.
+
+## Removing formulae
```

Comment:
Now that we also `deprecate!` and `disable!` formulae, it would be good to outline some guidelines here for doing that too.

Reply:
I'll probably just link to https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae